src/api/define.js

```diff
@@ -91,44 +94,26 @@ function createInitProps(Ctor) {
   };
 }
 
-function generateUniqueName(name) {
-  // http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript/2117523#2117523
-  const rand = 'xxxxxxxx'.replace(/[xy]/g, (c) => {
-    const r = Math.random() * 16 | 0;
-    const v = c === 'x' ? r : (r & 0x3 | 0x8); // eslint-disable-line no-mixed-operators
-    return v.toString(16);
-  });
-
-  return `${name}-${rand}`;
-}
-
-function prepareForRegistration(name, Ctor) {
-  Ctor[$name] = name;
+function prepareForRegistration(name, Ctor, { unique } = {}) {
   Ctor[$props] = createInitProps(Ctor);
   Ctor[$renderer] = createRenderer(Ctor);
+  Ctor.id = name;
+  Ctor.uniqueId = Ctor[$name] = unique ? `${name}-${rand()}` : name;
 }
 
 function registerV0Element(name, Ctor) {
-  let res;
-  let uniqueName;
   try {
     prepareForRegistration(name, Ctor);
-    res = document.registerElement(name, Ctor);
+    return document.registerElement(Ctor.uniqueId, Ctor);
   } catch (e) {
```

Comment:
Yeah, you can test the error message Blink gives you. This should be fairly innocuous as Blink isn't changing the old implementation. Should we do this as part of this PR? This is remnant from older changes.